eng/pipelines/templates/jobs/ci.yml

```diff
@@ -44,14 +44,17 @@ jobs:
           ArtifactName: 'packages'
 
     steps:
-      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
-        parameters:
-          Paths:
-            - 'sdk/${{ parameters.ServiceDirectory }}'
-            - '**/*.xml'
-            - '**/*.md'
-            - '!sdk/**/test-recordings'
-            - '!sdk/**/session-records'
+      # Skip sparse checkout for the `azure-sdk-for-<lang>-pr` private mirrored repositories
+      # as we require the github service connection to be loaded.
+      - ${{ if not(contains(variables['Build.DefinitionName'], 'java-pr')) }}:
```

Comment:
This assumes that the build definition is named appriopriately to show it s is using a private repo which might not always be the case. It might be better to use a condition in the sparse checkout steps. Anyways this works for normal cases so I will go ahead and merge it.

Reply:
Yes it's not ideal and relies on our pipeline generation naming to be consistent. I'll see if I can do it as a condition when I move the functionality into tools.